python/pyarrow/dataset.py

```diff
@@ -239,15 +240,18 @@ def _ensure_filesystem(fs_or_uri):
                 )
             filesystem = SubTreeFileSystem(prefix, filesystem)
         return filesystem, is_local
-    elif isinstance(fs_or_uri, (LocalFileSystem, _MockFileSystem)):
-        return fs_or_uri, True
-    elif isinstance(fs_or_uri, FileSystem):
-        return fs_or_uri, False
-    else:
+
+    try:
+        filesystem = _ensure_filesystem(fs_or_uri)
```

Comment:
It's a bit confusing to have those two functions named the same. Can you rename one or other to distinguish them better?

Reply:
Renamed the one in dataset.py